es-vue-base/src/lib-components/EsVerificationCode.vue

```diff
@@ -119,89 +144,81 @@ export default {
                 // Move to next input if possible
                 currentActiveElement.nextElementSibling?.focus();
             }
 
-            // If this is a paste and we stored the data parse it
-            if (inputType === 'insertFromPaste' && this.dataFromPaste) {
-                // Loop over each character in our paste
-                this.dataFromPaste.forEach((char) => {
-                    // Getting the current index handles case where user pastes into an input other than input[0]
-                    const curIndex = this.getElementIndex(currentActiveElement);
-                    this.code[curIndex] = char;
-
-                    // If the paste is less than this.charCount; move the focus to the first empty input
-                    // If the paste is equal or larger than this.charCount we move focus to the last input
-                    if (currentActiveElement.nextElementSibling) {
-                        // TODO: Why does chromium require setting the value directly?
-                        currentActiveElement.value = char;
-
-                        currentActiveElement = currentActiveElement.nextElementSibling;
-                        currentActiveElement?.focus();
-                    }
-                });
-            }
+            this.emitCodeUpdate();
         },
-        // Test pasting content
         handlePaste(event) {
-            const currentActiveElement = event.target;
-            const prevIndex = this.getElementIndex(currentActiveElement);
-
-            // TODO: Clearing this is not ideal but pasting over existing data doesn't work
-            this.code = Array(this.charCount);
-
             // Get the paste from clipboard; truncate it if its too long
             this.dataFromPaste = event.clipboardData?.getData('text')
-                .trim().split('').splice(0, this.charCount - prevIndex);
+                .trim().split('').splice(0, this.charCount);
 
-            // Check the data only contains accepted characters
-            const checkPasteValidity = this.checkCodeValidity(this.dataFromPaste);
+            // Reset the code to allow new paste
+            this.code = Array(this.charCount).fill('');
+            this.dataFromPaste.forEach((char, index) => {
+                this.code[index] = char;
+            });
 
-            // If the data doesn't pass validation don't update
-            if (!checkPasteValidity) {
-                event.preventDefault();
+            let focusIndex = this.dataFromPaste.length;
+            if (focusIndex > this.code.length - 1) {
+                focusIndex = this.code.length - 1;
             }
+
+            // Move focus to the next empty spot or last input
+            this.$refs[`codeInput_${focusIndex}`]?.[0]?.focus();
+
+            this.emitCodeUpdate();
         },
-        checkCodeValidity(code) {
-            if (!code) {
-                return false;
+        // Normalize strings into a valid array
+        stringToArray(value) {
```

Comment:
Similar to above, what do you think about always returning a fixed length based on `carCount`? Something like:

```js
stringToArray: (value) => Array(this.charCount).fill('').map((cur, index) => value[index] ? value[index] : cur);
```